### PR TITLE
[ts-sdk] Export more types

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
There were a lot of types that were not being exported, and many users asked for them to be exported.  Any types that are not exported here are exported under `TransactionBuilderTypes`

Additionally, I cleaned up a few interfaces that were either not using the already defined types, or were less permissive than other places that took in the same values.

### CODE REVIEW

File: ecosystem/typescript/sdk/src/index.ts [hunk 1]
```diff
@@ -4,17 +4,14 @@
...
 export * as Types from "./generated/index";
 export * from "./client";
-export { derivePath } from "./utils/hd-key";
-export {
-  deserializePropertyMap,
-  deserializeValueBasedOnTypeTag,
-  getPropertyValueRaw,
-} from "./utils/property_map_serde";
-export { Network, CustomEndpoints } from "./utils/api-endpoints";
+export * from "./utils/api-endpoints";
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/index.ts:13]: why not `export * from "./utils"`?

File: ecosystem/typescript/sdk/src/providers/index.ts [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
+
+/**
+ * Controls the number of results that are returned and the starting position of those results.
+ * limit specifies the maximum number of items or records to return in a query result.
+ * offset parameter specifies the starting position of the query result within the set of data.
+ * For example, if you want to retrieve records 11-20,
+ * you would set the offset parameter to 10 (i.e., the index of the first record to retrieve is 10)
+ * and the limit parameter to 10 (i.e., the number of records to retrieve is 10))
+ */
+export interface PaginationArgs {
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/providers/index.ts:15]: can probably remove it from here? as we now have `IndexerPaginationArgs`

**Review by 0xmaayan — APPROVED** [review]
noice!

File: ecosystem/typescript/sdk/src/providers/index.ts (hunk 2 above)
Comment (gregnazario) [reply ecosystem/typescript/sdk/src/providers/index.ts:15]: Yes, let me fix that.

Comment (gregnazario) [reply ecosystem/typescript/sdk/src/providers/index.ts:15]: For future versions, let's make one that's common, and do the name conversion for the user.

File: ecosystem/typescript/sdk/src/index.ts (hunk 1 above)
Comment (gregnazario) [reply ecosystem/typescript/sdk/src/index.ts:13]: There's some circular dependency

**Review by xbtmatt — APPROVED** [review]

**Pull request merged**